public/docs/ts/latest/guide/structural-directives.jade

```diff
@@ -68,57 +110,111 @@ figure.image-display
   All of them are destroyed when `ngIf` destroys the common ancestor.
   This cleanup effort is usually a good thing.
 
+  Les composants ont souvent des composants enfants qui ont eux-mêmes des enfants.
+  Chacun d’entre eux sont détruit lorsque `ngIf` détruit leur ancêtre commun.
+  Cet effort de nettoyage est généralement une bonne chose.
+
   Of course it isn't *always* a good thing.
   It might be a bad thing if we need that particular component again soon.
 
+  Bien sûr il n’est pas *toujours* une bonne chose.
+  Il serait peut-être un mauvais choix si nous avons rapidement besoin de ce composant particulier de nouveau.
+
   The component's state might be expensive to re-construct.
   When `ngIf` becomes `true` again, Angular recreates the component and its subtree.
   Angular runs every component's initialization logic again. That could be expensive ... as when
   a component re-fetches data that had been in memory just moments ago.
+
+  L'état du composant peut être coûteux à reconstruire.
+  Lorsque `ngIf` devient `true` de nouveau, Angular recrée le composant et son sous-arbre.
+  Angular lance chaque logique d'initialisation de chaque composant une nouvelle fois.
+  Cela pourrait être coûteux... comme quand un composant re-récupère des données qui étaient en mémoire il y a quelques instants.
 .l-sub-section
   :marked
     *Design thought*: minimize initialization effort and consider caching state in a
     companion service.
+
+    *Idée de conception* : minimiser l'effort d'initialisation et envisager la mise en cache des états
+    ​​dans un service dédié.
 :marked
   Although there are pros and cons to each approach,
   in general it is best to use `ngIf` to remove unwanted components rather than
   hide them.
 
+  Bien qu'il existe des avantages et des inconvénients à chaque approche,
+  en général, il est préférable d'utiliser `ngIf` pour éliminer les composants indésirables
+  plutôt que de les cacher.
+
   **These same considerations apply to every structural directive, whether built-in or custom.**
   We should ask ourselves &mdash; and the users of our directives &mdash; to think carefully
   about the consequences of adding and removing elements and of creating and destroying components.
 
+  **Ces mêmes considérations valent pour chaque directive structurelle, que ce soit celles existantes ou les votres.**
+  Nous devrions nous demander &mdash; et les utilisateurs de nos directives &mdash; à réfléchir sur les conséquences
+  de l'ajout et de la suppression d’éléments, et la création et la destruction de composants.
+
   Let's see these dynamics at work. For fun, we'll stack the deck *against*
   our recommendation and consider a component called `heavy-loader` that
   ***pretends*** to load a ton of data when initialized.
 
+  Voyons voir ces dynamiques à l'œuvre.
+  Pour le plaisir, nous allons *fausser* nos recommandation et considérer un composant appelé `heavy-loader`
```

Comment:
stack the deck est une expression (http://idioms.thefreedictionary.com/stack+the+deck) que je pense qu'on peut traduire par "piper les dés" ou "pousser le bouchon"